crates/conjure_core/src/ast/pretty.rs

```diff
@@ -0,0 +1,42 @@
+//! Functions for pretty printing Conjure models.
+//!
+//! Most things can be pretty printed using `Display`; however some, notably collections
+//! can not, for example, Vec<Expression>
+
+use itertools::Itertools;
+
+use super::Expression;
+
+/// Pretty prints a `Vec<Expression>` as if it were a top level constraint list in a `such that`.
+///
+/// Each expression is printed on a new line, and expressions are delimited by commas.
+///
+/// For some input expressions A,B,C:
+/// ```text
+/// A,
+/// B,
+/// C
+/// ```
+///
+/// Each `Expression` is printed using its underlying `Display` implementation.
+pub fn pretty_expressions_as_top_level(expressions: &[Expression]) -> String {
+    expressions.iter().map(|x| format!("{}", x)).join(",\n")
+}
+
+/// Pretty prints a `Vec<Expression>` as if it were a conjunction.
+///
+/// For some input expressions A,B,C:
+///
+/// ```text
+/// (A /\ B /\ C)
+/// ```
+///
+/// Each `Expression` is printed using its underlying `Display` implementation.
+pub fn pretty_expressions_as_conjunction(expressions: &[Expression]) -> String {
```

Comment:
why is this better than showing the vec itself in the logs?

Reply:
When we debug print a vector, it debug prints it's elements. We want to use Display for the elements.

An alternative to this particular method could be to print each new top separated by commas but not new lines (i.e. as a vector), but i did it this way to be closer to correct essence.

Reply:
Vec<T> does not implement Display.